packages/verser/test/http-connection.spec.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-console */
+import { defer } from "@scramjet/utility";
+import test from "ava";
+import { request, Server } from "http";
+import { createServer } from "@scramjet/api-server";
+import { Verser, VerserClient, VerserConnection } from "../src";
+
+test("Connect VerserClient A to Verser B and send HTTP GET Request to VerserClient A", async (t) => {
```

Comment:
Perhaps you could extract the main portion of the test to a separate function and provide the server from outside, so you could test it with `http2`, `https` and `http`?

Reply:
definitely a good idea

Reply:
i think there's too many use case specific details (we not only use http/https servers but also clients) in each test to completely extract it. I cleaned it up a bit by extracting some common steps to util functions.